src/main/java/KikuBot.java

```diff
@@ -0,0 +1,199 @@
+import exceptions.KikuEmptyTaskException;
+import exceptions.KikuException;
+import exceptions.KikuInvalidTaskException;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+public class KikuBot {
+
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static final String HORIZONTAL = "____________________________________________________________";
+    private static final String BOT_NAME = "Kiku";
+    private static final String FILE_NAME = "./data/Kiku.txt";
+
+    private static void saveTasks() {
+        try {
+            File file = new File(FILE_NAME);
+            File parentDir = file.getParentFile();
+
+            if (!parentDir.exists()) {
+                parentDir.mkdirs();
+            }
+
+            FileWriter writer = new FileWriter(file);
+            for (Task task : tasks) {
+                writer.write(task.toFileFormat() + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving tasks.");
+            e.printStackTrace();
+        }
+    }
+
+    private static void loadTasks() {
+        File file = new File(FILE_NAME);
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                Task task = parseTask(line);
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("No saved tasks found. Starting with an empty list.");
+            new File(file.getParent()).mkdirs();
+        }
```

Comment:
Perhaps you could abstract out some parts of this code into another method to avoid deep nesting.